src/Details.js

```diff
@@ -12,55 +12,72 @@ if (typeof window !== 'undefined') {
   require('details-element-polyfill')
 }
 
-const DetailsReset = styled('details')`
+const StyledDetails = styled('details')`
   & > summary {
     list-style: none;
   }
   & > summary::-webkit-details-marker {
     display: none;
   }
+
+  ${COMMON}
 `
 function getRenderer(children) {
   return typeof children === 'function' ? children : () => children
 }
 
-function DetailsBase({children, overlay, render = getRenderer(children), defaultOpen = false, ...rest}) {
-  const [open, setOpen] = useState(defaultOpen)
+function Details({
+  children,
+  overlay,
+  render = getRenderer(children),
+  open: userOpen,
+  onClickOutside,
+  onToggle,
+  defaultOpen = false,
+  ...rest
+}) {
+  const [internalOpen, setInternalOpen] = useState(defaultOpen)
   const ref = useRef(null)
+  // only use internal open state if user doesn't provide a value for the open prop
+  const open = userOpen || internalOpen
```

Comment:
Should this be
`const open = userOpen !== undefined ? userOpen || internalOpen`

Thinking of the case where the user has explicitly passed `false`, if `internalOpen` is true, won't this us as open?

Reply:
For example:
```
<State default={false}>
  {([open, setOpen]) => {
    const handleToggle = (e) => setOpen(e.target.open)
    const handleClickOutside = () => setOpen(false)

    return (
      <Details open={open} onToggle={handleToggle} onClickOutside={handleClickOutside} overlay>
        <Button as="summary">Click me</Button>
        <p onClick ={() => setOpen(false)}>This should show and hide</p>
      </Details>
    )
  }}
</State>
```

Doesn't close the Details when the `<p>` text is clicked, even though the user is now explicitly passing false, because the `internalOpen` is still true.

Reply:
good catch 🙌